packages/ui/lib/components/Display/Address.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect } from 'react'
+import {
+  FiCopy as CopyIcon,
+  FiExternalLink as ExternalLinkIcon,
+} from 'react-icons/fi'
+import useEns from '../../../../../unlock-app/src/hooks/useEns'
+import { Placeholder } from '../Placeholder'
+import useClipboard from 'react-use-clipboard'
+import { ToastHelper } from '../../../../../unlock-app/src/components/helpers/toast.helper'
+import { minifyAddress } from '~/utils'
+
+export interface AddressProps {
+  address: string
+  showExternalLink?: boolean
+  showCopyIcon?: boolean
+  showENSName?: boolean
+  className?: string
+  minified?: boolean
+}
+
+// Address component
+export const Address: React.FC<AddressProps> = ({
+  address,
+  showExternalLink = false,
+  showCopyIcon = false,
+  showENSName = true,
+  className = '',
+  minified = true,
+}) => {
+  const resolvedName = useEns(address)
```

Comment:
I think we should have have a `useName` prop passed to the component here that would resolve. This way the app can set its own resolution.